derive_client/clients/base_client.py

```diff
@@ -163,10 +163,10 @@ def deposit_to_derive_result(
         amount = int(amount * 10 ** TOKEN_DECIMALS[UnderlyingCurrency[currency.name.upper()]])
         client = BridgeClient(self.env, chain_id, account=self.signer, wallet=self.wallet, logger=self.logger)
 
-        if currency == Currency.DRV:
-            return client.deposit_drv(amount=amount, currency=currency)
+        prepared_tx = client.prepare_deposit(amount=amount, currency=currency)
+        tx_result = client.submit_bridge_tx(prepared_tx)
 
-        return client.deposit(amount=amount, currency=currency)
+        return client.poll_bridge_progress(tx_result=tx_result)
```

Comment:
more advanced user would likely want access to the following methods on the `BridgeClient`:
- `prepare_deposit` (so they can inspect the transaction gas cost and correctness)
- `prepare_withdrawal` (ditto)
- `submit_bridge_tx` (so they can bump gas on `PreparedBridgeTx`, re-sign, and resubmit for replacement)
- `confirm_source_tx` (more fine grained control on tx finality, if we allow passing `TxFinalityParams`) 
- `wait_for_target_event` (ditto)
- `confirm_target_tx` (ditto)